src/test/java/de/tum/in/www1/artemis/AchievementIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+        achievement.setDescription("Create correct relations");
+        achievement.setIcon("");
+        achievement.setRank(1);
+        achievement.setCourse(course);
+
+        achievement = achievementRepository.save(achievement);
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    public void testManyToManyRelationToUser() {
+        assertThat(user.getAchievements().size()).isEqualTo(0).as("Number of achievements for user should be 0");
+
+        user.addAchievement(achievement);
+        assertThat(user.getAchievements().size()).isEqualTo(1).as("Number of achievements for user should be 1");
+        assertThat(user.getAchievements().contains(achievement)).isTrue().as("User has correct achievement");
+        assertThat(achievement.getUsers().size()).isEqualTo(1).as("Number of users for achievement should be 1");
+        assertThat(achievement.getUsers().contains(user)).isTrue().as("Achievement has correct user");
+
+        user.removeAchievement(achievement);
+        assertThat(user.getAchievements().size()).isEqualTo(0).as("Number of achievements for user should be 0");
+        assertThat(user.getAchievements().contains(achievement)).isFalse().as("User does not have removed achievement");
+        assertThat(achievement.getUsers().size()).isEqualTo(0).as("Number of users for achievement should be 0");
+        assertThat(achievement.getUsers().contains(user)).isFalse().as("Achievement does not have incorrect user");
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testManyToManyRelationToUserRepository() throws Exception {
+        user.addAchievement(achievement);
+        user = userRepository.save(user);
+
+        var achievements = request.get("/api/achievements", HttpStatus.OK, Set.class);
+        assertThat(achievements.size()).isEqualTo(1).as("Number of achievements for user should be 1");
+
+        userRepository.delete(user);
+        assertThat(achievementRepository.findAll().contains(achievement)).isTrue().as("Achievement does not get deleted if user does");
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testManyToOneRelationToCourseRepository() throws Exception {
+        var achievements = request.get("/api/courses/" + course.getId() + "/achievements", HttpStatus.OK, Set.class);
+        assertThat(achievements.size()).isEqualTo(1).as("Number of achievements for course should be 1");
+
+        courseRepository.delete(course);
+        assertThat(achievementRepository.findAll().contains(achievement)).isFalse().as("Achievement gets deleted if course does");
+    }
+
+    @Test
+    public void testUpdateAchievement() {
+        user.addAchievement(achievement);
+        userRepository.save(user);
+        var achievementToUpdate = achievementRepository.findById(achievement.getId());
+        if (achievementToUpdate.isPresent()) {
```

Comment:
I think this test will be also successful even though the `achievementToUpdate.isPresent()` is `false`

Reply:
Removed the if statement :) thanks